tlb/hashmap.go

```diff
@@ -12,6 +13,7 @@ import (
 type fixedSize interface {
 	FixedSize() int
 	Equal(other any) bool
+	Compare(other any) int
```

Comment:
does it make sense to change the method to:
```
Compare(other any) (int, bool)
or 
Compare (other any) (int, error)
```
because returning 0 when we can't cast `other` to a specific type looks unreliable